x/tokenomics/keeper/settle_session_accounting.go

```diff
@@ -31,64 +29,72 @@ const (
 // against a proof BEFORE calling this function.
 //
 // TODO_BLOCKER(@Olshansk): Is there a way to limit who can call this function?
-func (k Keeper) SettleSessionAccounting(ctx context.Context, claim *prooftypes.Claim) error {
-	// Parse the context
+func (k Keeper) SettleSessionAccounting(
+	ctx context.Context,
+	claim *prooftypes.Claim,
+) error {
 	logger := k.Logger().With("method", "SettleSessionAccounting")
 
+	// Make sure the claim is not nil
 	if claim == nil {
 		logger.Error("received a nil claim")
 		return types.ErrTokenomicsClaimNil
 	}
 
+	// Retrieve & validate the session header
 	sessionHeader := claim.GetSessionHeader()
-
-	// Make sure the session header is not nil
 	if sessionHeader == nil {
 		logger.Error("received a nil session header")
 		return types.ErrTokenomicsSessionHeaderNil
 	}
-
-	// Validate the session header
 	if err := sessionHeader.ValidateBasic(); err != nil {
 		logger.Error("received an invalid session header", "error", err)
 		return types.ErrTokenomicsSessionHeaderInvalid
 	}
 
-	// Decompose the claim into its constituent parts for readability
 	supplierAddr, err := sdk.AccAddressFromBech32(claim.GetSupplierAddress())
-	if err != nil {
+	if err != nil || supplierAddr == nil {
 		return types.ErrTokenomicsSupplierAddressInvalid
 	}
 
 	applicationAddress, err := sdk.AccAddressFromBech32(sessionHeader.GetApplicationAddress())
-	if err != nil {
+	if err != nil || applicationAddress == nil {
 		return types.ErrTokenomicsApplicationAddressInvalid
 	}
 
-	// Retrieve the application
-	application, found := k.applicationKeeper.GetApplication(ctx, applicationAddress.String())
-	if !found {
-		logger.Error(fmt.Sprintf("application for claim with address %s not found", applicationAddress))
-		return types.ErrTokenomicsApplicationNotFound
-	}
-
+	// Retrieve the sum of the root as a proxy into the amount of work done
 	root := (smt.MerkleRoot)(claim.GetRootHash())
 
 	// TODO_DISCUSS: This check should be the responsibility of the SMST package
 	// since it's used to get compute units from the root hash.
-	if len(root) != smstRootSize {
+	if root == nil || len(root) != smstRootSize {
 		logger.Error(fmt.Sprintf("received an invalid root hash of size: %d", len(root)))
 		return types.ErrTokenomicsRootHashInvalid
 	}
-
-	// Retrieve the sum of the root as a proxy into the amount of work done
 	claimComputeUnits := root.Sum()
 
+	// Helpers for logging the same metadata throughout this function calls
+	logger = logger.With(
+		"compute_units", claimComputeUnits,
+		"session_id", sessionHeader.GetSessionId(),
+		"supplier", supplierAddr,
+		"application", applicationAddress,
+	)
+
+	logger.Info("About to start session settlement accounting")
+
+	// Retrieve the staked application record
+	application, foundApplication := k.applicationKeeper.GetApplication(ctx, applicationAddress.String())
+	if !foundApplication {
+		logger.Error(fmt.Sprintf("application for claim with address %s not found", applicationAddress))
```

Comment:
I would propose making this a warn level for consistency with other places where we're logging and returning the same error. IMO these are targets for https://github.com/pokt-network/poktroll/pull/448#discussion_r1549742985 as well.

```suggestion
		logger.Warn(fmt.Sprintf("application for claim with address %q not found", applicationAddress))
```